lucene/core/src/java/org/apache/lucene/search/FuzzyQuery.java

```diff
@@ -183,7 +162,7 @@ public void visit(QueryVisitor visitor) {
       if (maxEdits == 0 || prefixLength >= term.text().length()) {
         visitor.consumeTerms(this, term);
       } else {
-        automata[automata.length - 1].visit(visitor, this, field);
+        visitor.consumeTermsMatching(this, term.field(), () -> getAutomata().runAutomaton);
```

Comment:
Does this still build the automata each time?

Reply:
Only if the visitor implementation actually needs it, we're passing a `Supplier` now.